File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/stored/library/items/files/retrieval/GetStoredFiles.java
package com.lasthopesoftware.bluewater.client.stored.library.items.files.retrieval;

import com.lasthopesoftware.bluewater.client.browsing.library.items.media.files.ServiceFile;
import com.lasthopesoftware.bluewater.client.browsing.library.repository.LibraryId;
import com.lasthopesoftware.bluewater.client.stored.library.items.files.repository.StoredFile;
import com.namehillsoftware.handoff.promises.Promise;

public interface GetStoredFiles {
	Promise<StoredFile> promiseStoredFile(LibraryId libraryId, ServiceFile serviceFile);
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/stored/library/items/files/retrieval/GetStoredFiles.kt
package com.lasthopesoftware.bluewater.client.stored.library.items.files.retrieval

import com.lasthopesoftware.bluewater.client.browsing.files.ServiceFile
import com.lasthopesoftware.bluewater.client.browsing.library.repository.LibraryId
import com.lasthopesoftware.bluewater.client.stored.library.items.files.repository.StoredFile
import com.namehillsoftware.handoff.promises.Promise

interface GetStoredFiles {
    fun promiseStoredFile(libraryId: LibraryId, serviceFile: ServiceFile): Promise<StoredFile>
}
